src/librustdoc/html/static/search.js

```diff
@@ -988,19 +988,43 @@ window.initSearch = function(rawSearchIndex) {
                 }
                 length += 1;
 
-                output += "<a class=\"result-" + type + "\" href=\"" + item.href + "\">" +
-                          "<div><div class=\"result-name\">" +
-                          (item.is_alias === true ?
-                           ("<span class=\"alias\"><b>" + item.alias + " </b></span><span " +
-                              "class=\"grey\"><i>&nbsp;- see&nbsp;</i></span>") : "") +
-                          item.displayPath + "<span class=\"" + type + "\">" +
-                          name + "</span></div><div>" +
-                          "<span class=\"desc\">" + item.desc +
-                          "&nbsp;</span></div></div></a>";
+                var link = document.createElement("a");
+                link.className = "result-" + type;
+                link.href = item.href;
+
+                var wrapper = document.createElement("div");
+                var resultName = document.createElement("div");
+                resultName.className = "result-name";
+
+                if (item.is_alias) {
+                    var alias = document.createElement("span");
+                    alias.className = "alias";
+
+                    var bold = document.createElement("b");
+                    bold.innerText = item.alias;
+                    alias.appendChild(bold);
+
+                    alias.innerHTML += "<span class=\"grey\"><i>&nbsp;- see&nbsp;</i></span>";
```

Comment:
Afaik `innerhtml += ...` is bad for performance. [insertAdjacentHtml](https://developer.mozilla.org/en-US/docs/Web/API/Element/insertAdjacentHTML) should be used instead.

Reply:
Yes, but touching the DOM is even worse for performance so it is a tradeoff between attaching all Results once and a little less optimal manipulation
> Note: The beforebegin and afterend positions work only if the **node is in the DOM tree** and **has a parent** element.

Reply:
This is about *inner*HTML, which you would replace with `afterbegin` or `beforeend`. You would only use `beforebegin` or `afterend` if you wanted to replace *outer*HTML.

Reply:
True, you are right!